extensions/kubectl-cli/src/extension.ts

```diff
@@ -337,7 +337,44 @@ async function postActivate(
   let releaseToInstall: KubectlGithubReleaseArtifactMetadata | undefined;
   let releaseVersionToInstall: string | undefined;
   let currentVersion = kubectl.version;
-
+  // check if there is a new version to be installed and register the updater
+  let releaseToUpdateTo: KubectlGithubReleaseArtifactMetadata | undefined;
+  let releaseVersionToUpdateTo: string | undefined;
+  const latestAsset = await kubectlDownload.getLatestVersionAsset();
+  const update = {
+    version: latestAsset.tag.slice(1) !== kubectlCliTool.version ? latestAsset.tag.slice(1) : undefined,
+    selectVersion: async (): Promise<string> => {
+      const selected = await kubectlDownload.promptUserForVersion(currentVersion);
+      releaseToUpdateTo = selected;
+      releaseVersionToUpdateTo = selected.tag.slice(1);
+      return releaseVersionToUpdateTo;
+    },
+    doUpdate: async (): Promise<void> => {
+      if (!releaseToUpdateTo || !releaseVersionToUpdateTo) {
```

Comment:
In case of update, the version being installed is defined in 'version' attribute above and 'selectVerison' function call is going to be skipped on UI side, hence `releaseToUpdateTo` and `releaseVersionToUpdateTo` are empty and that the means the update is requested.